src/plugins/jma/datasource.py

```
@@ -0,0 +1,383 @@
+                return
+            
+            current_date = start_date or datetime.now()
+            end = end_date or datetime.now()
+            
+            measurements_count = 0
+            
+            # AMeDAS provides data in 10-minute intervals
+            while current_date <= end and (not limit or measurements_count < limit):
+                # Round to nearest 10 minutes
+                minutes = (current_date.minute // 10) * 10
+                timestamp_str = current_date.replace(minute=minutes, second=0, microsecond=0).strftime('%Y%m%d%H%M%S')
```

Pylint (3.3.7)

[convention] Line too long (118/100)

(C0301)